### ISSUE TITLE
User registration does not work with Email verification + Octane Server

### ISSUE THREAD

**Issue by miguilimzero** [laravel-filament/filament#7922]
### Package

filament/filament

### Package Version

v3.0.27

### Laravel Version

10.19.0

### Livewire Version

v3.0.0-beta.9

### PHP Version

PHP 8.2.9

### Problem description

An exception occurs when registering an account using the Filament register page + `MustVerifyEmail` + octane server.

### Expected behavior

The registration works normally with the verification email sent and redirected to the verify email page.

### Steps to reproduce

- Clone example repository
- Setup .env variables
- php artisan migrate
- php artisan octane:start (Important! Bug only happens with octane server)
- Try to register an account using the register page `admin/register`

### Reproduction repository

https://github.com/miguilimzero/filament-email-verification-bug-example

### Relevant log output

![image](https://github.com/filamentphp/filament/assets/35383529/9ca08a5d-299e-4ea5-93e3-474e41829bdf)

**Comment by miguilimzero** [comment]
It seems the following binding at `Filament\Pages\Auth\Register` -> `register()`  method does not actually works with Octane server, it just not being applied:

```php
        app()->bind(
            \Illuminate\Auth\Listeners\SendEmailVerificationNotification::class,
            \Filament\Listeners\Auth\SendEmailVerificationNotification::class,
        );
``` 

This could be easily fixed by moving this `bind()` to the `FilamentServiceProvider`, however, this would overwrite the email notification listener for the entire application at any request, which I think is not the intended behavior by Filament.

**Comment by danharrin** [comment]
Yeah, I'm not sure how to handle this either. Do you think it might be a bug in Laravel?

**Comment by meeftah** [comment]
is there any progress with this error?

**Comment by miguilimzero** [comment]
@danharrin I believe there are some things to point out in this issue:

- Laravel Octane boots the application only once (when it starts the server).

- What would be the expected behavior of this code with Octane? This code should bind the verification email on the first request and persist to all others.

- The likely bug is that it never binds the verification email. Even if this likely bug gets fixed, it would still not be the intended behavior by Filament. 

- We just can't bind something per request using Octane. (Maybe Octane offers a listener to reset an app binding on every request, so that works, but by default, we can't).

I would suggest removing the binding and creating a custom event `FilamentRegistered` instead of using the default one. But unfortunately, this would be a breaking change.

I will try to search if there is something we can do with Octane listeners to get around this problem.

**Comment by danharrin** [comment]
Fixed by #9875, I have written notes to detail the solution
